src/Tizen.NUI/src/internal/Layouting/RelativeLayout.cs

```diff
@@ -0,0 +1,280 @@
+
+namespace Tizen.NUI
+{
+    // Methods and Properties for calculation.
+    public partial class RelativeLayout
+    {
+        readonly static Dictionary<View, Relative> HorizontalRelativeCache = new Dictionary<View, Relative>();
+        readonly static Dictionary<View, Relative> VerticalRelativeCache = new Dictionary<View, Relative>();
+        private static readonly RelativePropertyGetters RelativeHorizontalPropertyGetters = new RelativePropertyGetters
+        {
+            RelativeCache = HorizontalRelativeCache,
+            GetTargetFrom = GetLeftTarget,
+            GetTargetTo = GetRightTarget,
+            GetRelativeOffsetFrom = GetLeftRelativeOffset,
+            GetRelativeOffsetTo = GetRightRelativeOffset,
+            GetAlignment = GetHorizontalAlignment,
+            GetFill = GetFillHorizontal,
+            GetMeasuredSize = (item) => { return item.MeasuredWidth.Size.AsDecimal(); }
+        };
+        private static readonly RelativePropertyGetters RelativeVerticalPropertyGetters = new RelativePropertyGetters
+        {
+            RelativeCache = VerticalRelativeCache,
+            GetTargetFrom = GetTopTarget,
+            GetTargetTo = GetBottomTarget,
+            GetRelativeOffsetFrom = GetTopRelativeOffset,
+            GetRelativeOffsetTo = GetBottomRelativeOffset,
+            GetAlignment = GetVerticalAlignment,
+            GetFill = GetFillVertical,
+            GetMeasuredSize = (item) => { return item.MeasuredHeight.Size.AsDecimal(); }
+        };
+
+        delegate Relative GetRelative(View view, in float layoutSize);
+
+        private bool ValidateRelative(Relative relative, in float layoutSize, bool isFixedRelative)
+        {
+            float spaceSize = relative.spaceGeometry.Size;
+            float viewPosion = relative.viewGeometry.Position;
+            float viewSize = relative.viewGeometry.Size;
+            (float start, float end) space = (relative.spaceGeometry.Position, relative.spaceGeometry.Position + spaceSize);
+
+            if (spaceSize > float.Epsilon)
+            {
+                if (!isFixedRelative && spaceSize < viewSize)
+                    return false;
+
+                if (space.start < 0 && space.end <= layoutSize)
+                {
+                    if (viewPosion + viewSize > layoutSize)
+                        return false;
+                }
+                else if (0 <= space.start && layoutSize < space.end)
+                {
+                    if (viewPosion < 0)
+                        return false;
+                }
+                else if (viewPosion < 0 || viewPosion + viewSize > layoutSize)
+                {
+                    return false;
+                }
+            }
+            else if (space.start == 0.0)
+            {
+                if (viewPosion + viewSize > layoutSize)
+                    return false;
+            }
+            else if (space.start == layoutSize)
+            {
+                if (viewPosion < 0)
```

Comment:
If viewPosion + viewSize is bigger than layoutSize with space.start is equal to layoutSize, then is it returning true case?
```
else if (space.start == layoutSize)
{
    ....
    if (viewPosion + viewSize > layoutSize) return true;
}
```

Reply:
In that case, it will return `true` on line 98.

Reply:
Oh, sorry I think I asked unclearly;;

I thought that if viewPosition + viewSize is bigger than layoutSize, then the layoutSize should become bigger than now so the return value should be false regardless of the value of space.start.

Reply:
Thank you for explanation!

As I understand, if view starts from the left end of layout and a part of view is outside of layout's left side due to view alignment (bigger than 0), then that view area cannot be covered by increasing layoutSize. So layoutSize is not increased for this case and the view area outside of layout's left side is ignored.

In the same way, if view starts from the right end of layout and a part of view is outside of layout's right side due to view alignment (less than 1), then that view area cannot be covered by increasing layoutSize. So layoutSize is not increased for this case and the view area outside of layout's right side is ignored.

Thank you! :)